### PR TITLE
feat(full_url_for): bring up a new helper

### PR DESCRIPTION
**Pull request by SukkaW**
https://github.com/hexojs/hexo/pull/3701

### CODE REVIEW

**Comment by SukkaW** [comment]
The test will passed after #83 is fixed.

**Comment by curbengh** [comment]
The test pass now, just need to remove `console.log` and line ending.

File: lib/full_url_for.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+function fullUrlForHelper(path = '/') {
+  if (path.startsWith('//')) return path;
+  const { config } = this;
+  const data = parse(path);
+
+  // Exit if this is an external path
+  if (data.protocol) return path;
+
+  // Prepend root path
```
Comment (curbengh) [inline lib/full_url_for.js]: I don't think this comment is necessary, the next line is obvious on what it does.

File: test/full_url_for.spec.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+describe('full_url_for', () => {
+  const ctx = {
+    config: {}
+  };
+
+  const fullUrlFor = require('../lib/full_url_for').bind(ctx);
+
+  it('internal url - root directory', () => {
+    ctx.config.root = '/';
```
Comment (curbengh) [inline test/full_url_for.spec.js]: `config.root` is never used, so there's no need to test it.

File: test/full_url_for.spec.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  it('internal url - subdirectory', () => {
+    ctx.config.root = '/blog/';
+    ctx.config.url = 'https://example.com/blog';
+    fullUrlFor('index.html').should.eql(ctx.config.url + '/index.html');
+    fullUrlFor('/').should.eql(ctx.config.url + '/');
+  });
+
+  it('internal url - no duplicate slash', () => {
+    ctx.config.root = '/';
+    ctx.config.url = 'https://example.com';
```
Comment (curbengh) [inline test/full_url_for.spec.js]: `config.root` is never used, it should be `ctx.config.url = 'https://example.com/';`

File: lib/full_url_for.js (hunk 1 above)
Comment (SukkaW) [reply lib/full_url_for.js]: Removed.

File: test/full_url_for.spec.js (hunk 2 above)
Comment (SukkaW) [reply test/full_url_for.spec.js]: Removed.

File: test/full_url_for.spec.js (hunk 3 above)
Comment (SukkaW) [reply test/full_url_for.spec.js]: Removed.

**Pull request merged**